cdap-ui/app/cdap/components/FastAction/SetPreferenceAction/index.js

```diff
@@ -107,6 +107,7 @@ export default class SetPreferenceAction extends Component {
               toggleModal={this.toggleModal}
               entity={this.props.entity}
               onSuccess={this.onSuccess}
+              setAtNamespaceLevel={this.props.setAtNamespaceLevel}
```

Comment:
perhaps rather than a prop that is specific to a particular level, you can just have a `level` props, and pass in the level value. The component then can do with the information however they choose.